pkg/virt-controller/watch/vm.go

```diff
@@ -2375,3 +2386,68 @@ func (c *VMController) applyDevicePreferences(vm *virtv1.VirtualMachine, vmi *vi
 	}
 	return nil, nil
 }
+
+func (c *VMController) handleInterfaceRequests(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) error {
+	if len(vm.Status.InterfaceRequests) == 0 {
+		return nil
+	}
+
+	interfaceMap := map[string]virtv1.Interface{}
+	if vmi != nil {
+		interfaceMap = netvmispec.IndexInterfaceSpecByName(vmi.Spec.Domain.Devices.Interfaces)
+	}
+	for i, ifaceRequest := range vm.Status.InterfaceRequests {
+		vm.Spec.Template.Spec = *controller.ApplyNetworkInterfaceRequestOnVMISpec(
+			&vm.Spec.Template.Spec, &vm.Status.InterfaceRequests[i])
+
+		if vmi == nil || vmi.DeletionTimestamp != nil {
+			continue
+		}
+
+		if ifaceRequest.AddInterfaceOptions != nil {
+			if _, exists := interfaceMap[ifaceRequest.AddInterfaceOptions.InterfaceName]; exists {
+				continue
+			}
```

Comment:
> > While thinking about this, a question popped up: What will happen if the current interface is of a wrong type/binding? Do we cover that scenario?
> 
> We hard-code the binding in the only allowed caller - virtctl

This is the scenario I was referring to:
- The VM & VMI has an existing interface of binding FOO.
- Someone removed it from the VM spec.
- An interface hotplug is issued, persisted, with the same logical name, with a specific NAD name.

Because the binding is implicit "bridge", and the NAD will not get updated on the VMI, it seems we make have an unexpected behavior.
Do I understand this wrongly?